src/index.ts

```diff
@@ -70,3 +72,70 @@ export interface Location {
  * element are otherwise identical to the anchor element.
  */
 export const Link: React.ComponentClass<React.HTMLProps<{}>> = _Link
+
+/**
+ * Declare that a component is a controller.
+ *
+ * You must annotate a React Component as a controller before attaching services to it.
+ * You do not need to annotate a service before attaching other services to it.
+ */
+export function controller(): ComponentDecorator {
+  return (cls) => decorateController(cls)
+}
+
+/**
+ * Attach a service to a property of a controller or other service.
+ *
+ * You must annotate a React Component as a controller before attaching services to it.
+ * You do not need to annotate a service before attaching other services to it.
+ *
+ * ```
+ *  @controller
+ *  class Foo extends React.Component {
+ *    @service(MyService)
+ *    myService: MyService
+ *  }
+ * ```
+ */
+export function service(constructor: new () => Service): PropertyDecorator {
+  return decorateServiceProperty(constructor)
+}
+
+/**
+ * Base class for creating Services.
+ *
+ * Services are classes that have access to a subset of the React Component API.
+ * They are owned by a parent component, and receive lifecycle events.
+ *
+ * @class
+ */
+export interface Service<State = {}> {
```

Comment:
Will services need to respond to component updates?

Reply:
Probably. A graphql service might want to re-fetch on props changes, for example. But that operation would likely be asynchronous, so it would depend on the data fetching system being implemented.